lib/avalon/ffprobe.rb

```diff
@@ -26,35 +28,51 @@ def initialize(media_path)
         return
       end
       @json_output = JSON.parse(raw_output).deep_symbolize_keys
-      @video_stream = @json_output[:streams].select { |stream| stream[:codec_type] == 'video' }.first
+
+      @video_stream = json_output[:streams].select { |stream| stream[:codec_type] == 'video' }.first
+      @audio_stream = json_output[:streams].select { |stream| stream[:codec_type] == 'audio' }.first
     end
 
     def video?
-      # ffprobe treats plain text files as ANSI or ASCII art. This sets the codec type to video
-      # but leaves display aspect ratio `nil`. If display_aspect_ratio is nil, return false.
-      # ffprobe treats image files as a single frame video. This sets the codec type to video
-      # but the packet/frame count will equal 1. If packet count equals 1, return false.
-      return true if @video_stream && @video_stream[:display_aspect_ratio] && @video_stream[:nb_read_packets].to_i > 1
+      return true if video_stream
 
       false
     end
 
     def audio?
-      @json_output[:streams]&.any? { |stream| stream[:codec_type] == 'audio' }
+      return true if audio_stream
+
+      false
     end
 
     def duration
       return unless video? || audio?
       # ffprobe return duration as seconds. Existing Avalon logic expects milliseconds.
-      (@json_output[:format][:duration].to_f * 1000).to_i
+      (json_output[:format][:duration].to_f * 1000).to_i
     end
 
     def display_aspect_ratio
-      @video_stream[:display_aspect_ratio] if video?
+      if video? && video_stream[:display_aspect_ratio].present?
+        video_stream[:display_aspect_ratio]
+      elsif video?
+        video_stream[:width].to_f / video_stream[:height].to_f
+      end
     end
 
     def original_frame_size
-      "#{@video_stream[:width]}x#{@video_stream[:height]}" if video?
+      "#{video_stream[:width]}x#{video_stream[:height]}" if video?
+    end
+
+    private
+
+    def valid_content_type? media_file
+      # Remove S3 credentials or other params from extension output
+      extension = File.extname(media_file.location)&.gsub(/[\?#].*/, '')
+      # Fall back on file extension if magic bytes fail to identify file
+      content_type = Marcel::MimeType.for media_file.reader, extension: extension
+      return true if ['audio', 'video'].any? { |type| content_type.include?(type) }
+
+      false
```

Comment:
I think the conditional return is redundant.
```suggestion
      ['audio', 'video'].any? { |type| content_type.include?(type) }
```